_spinnaker/pacrd.md

```diff
@@ -0,0 +1,372 @@
+❯ kubectl get pipe myapplicationpipeline
+NAME                    STATUS    LASTCONFIGURED   URL
+myapplicationpipeline   Updated   5s               http://localhost:9000/#/applications/myapplicationname/executions/configure/f1eb82ce-5a8f-4b7a-9976-38e4aa022702
+
+# or...
+
+
+❯ kubectl get pipeline myapplicationpipeline
+NAME                    STATUS    LASTCONFIGURED   URL
+myapplicationpipeline   Updated   44s              http://localhost:9000/#/applications/myapplicationname/executions/configure/f1eb82ce-5a8f-4b7a-9976-38e4aa022702
+```
+
+A `describe` call will give you additional contextual information about the
+status of your pipeline:
+
+```
+❯ kubectl describe pipeline myapplicationpipeline
+Name:         myapplicationpipeline
+API Version:  pacrd.armory.spinnaker.io/v1alpha1
+Kind:         Pipeline
+Metadata:
+  # omitted for brevity
+Spec:
+  # omitted for brevity
+Status:
+  Id:               f1eb82ce-5a8f-4b7a-9976-38e4aa022702
+  Last Configured:  2020-03-09T15:55:27Z
+  Phase:            Updated
+  URL:              http://localhost:9000/#/applications/myapplicationname/executions/configure/f1eb82ce-5a8f-4b7a-9976-38e4aa022702
+Events:
+  Type     Reason                 Age                From       Message
+  ----     ------                 ----               ----       -------
+  Normal   Updated                94s                pipelines  Pipeline successfully created in Spinnaker.
+  Warning  ErrorUpdatingPipeline  93s                pipelines  Bad Request: The provided id f1eb82ce-5a8f-4b7a-9976-38e4aa022702 doesn't match the pipeline id null
+  Normal   Updated                91s (x2 over 91s)  pipelines  Pipeline successfully updated in Spinnaker.
+```
+
+### Updating
+
+Pipelines can be updated in one of two ways:
+
+1. Re-applying the pipeline manifest in your repository
+    - `kubectl apply -f pipeline.yaml`
+1. Editing the pipeline manifest in-cluster
+    - `kubectl edit pipeline myapplicationpipeline`
+
+In both cases, when a pipeline has been updated in your Kubernetes cluster,
+you will see changes propagate into Spinnaker shortly. If an error occurred
+during the update your pipeline may show an `ErrorFailedUpdate` state. You
+can see the details of that failure by describing the resource and looking in
+the "Events" section:
+
+```
+kubectl describe pipeline myapplicationpipeline
+```
+
+### Deleting
+
+When a pipeline is no longer necessary, you can delete it in one of two ways:
+
+1. Deleting the pipeline manifest from your repository definition
+  - `kubectl delete -f pipeline.yaml`
+1. Deleting the pipeline directly
+  - `kubectl delete pipeline myapplicationpipeline`
+
+In either case the pipeline will be deleted shortly in Spinnaker as well. If
+an error occurred during deletion then your pipeline may show an
+`ErrorFailedDelete` state. You can see the details of that failure by
+describing the resource and looking in the "Events section":
+
+```
+kubectl describe pipeline myapplicationpipeline
+```
+
```

Comment:
Should we include a final section here that essentially mimics the changelog of the service? (I believe Aimee was asking for this in a thread comment in Slack earlier today)

Reply:
Hmmm once we are at Early Release we can maintain a changelog. I dont think the overhead is worth it for an experiment

Reply:
Hmmm.... I don't remember suggesting a changelog, just a prerequisites section.

BTW really like that you included a Known Limitations section!

Reply:
np, we can leave it in the repo for now (private anywho).

Yeah, gotta make sure expectations are set for customers! ahah